src/middlewared/middlewared/plugins/pwenc.py

```diff
@@ -120,101 +115,22 @@ class Config:
-        ctdb_info = json.loads(Path(CTDB_VOL_INFO_FILE).read_text())
-        with glfs.get_volume_handle(ctdb_info['volume_name']) as vol:
-            secret_dir = vol.open_by_uuid(ctdb_info['uuid']).lookup('.cluster_private')
-            secret = secret_dir.lookup('clpwenc_secret')
-            self.secret = secret.contents()
+    secret_path = GlusterConfig.SECRETS_FILE.value
+    lock = threading.RLock()
 
     def _write_secret(self, secret, reset_passwords):
-        self._init_secret_dir()
-        with glfs.get_volume_handle(self.ctdb_info['volume_name']) as vol:
-            secret_dir = vol.open_by_uuid(self.secret_dir_uuid)
-            self._secret_permcheck(secret_dir, 0o700, True)
-            try:
-                secret_file = secret_dir.lookup('clpwenc_secret')
-            except pyglfs.GLFSError as e:
-                if e.errno != errno.ENOENT:
-                    raise
-                secret_file = secret_dir.create('clpwenc_secret', os.O_RDWR | os.O_CREAT)
-
-            self._secret_permcheck(secret_file, 0o600, False)
-            with lock_file_open(secret_file, os.O_RDWR) as fd:
-                fd.ftruncate(0)
-                fd.pwrite(secret, 0)
-                fd.fsync()
-
-    def _reset_passwords(self):
         raise NotImplementedError
 
-    @staticmethod
-    def _secret_opener(path, flags):
+    def generate_secret(self, reset_passwords=True):
+        # secret is derived from our pre-shared localevent secret
         raise NotImplementedError
 
-    def check(self):
-        self._init_secret_dir()
-        with glfs.get_volume_handle(self.ctdb_info['volume_name']) as vol:
-            secret_dir = vol.open_by_uuid(self.secret_dir_uuid)
-            self._secret_permcheck(secret_dir, 0o700, True)
-            try:
-                check_file = secret_dir.lookup('.check_file')
-
-            except pyglfs.GLFSError as e:
-                if e.errno != errno.ENOENT:
-                    raise
-
-                return False
-
-            return self.decrypt(check_file.contents()) == PWENC_CHECK
-
-    def _reset_pwenc_check_field(self):
-        self._init_secret_dir()
-        with glfs.get_volume_handle(self.ctdb_info['volume_name']) as vol:
-            secret_dir = vol.open_by_uuid(self.secret_dir_uuid)
-            self._secret_permcheck(secret_dir, 0o700, True)
-            try:
-                check_file = secret_dir.lookup('.check_file')
-
-            except pyglfs.GLFSError as e:
-                if e.errno != errno.ENOENT:
-                    raise
-                check_file = secret_dir.create('.check_file', os.O_RDWR)
+    def _reset_passwords(self):
+        raise NotImplementedError
 
-            with lock_file_open(check_file, os.O_RDWR) as fd:
-                fd.ftruncate(0)
-                fd.pwrite(self.encrypt(PWENC_CHECK), 0)
+    def _read_secret(self):
+        with open(self.secret_path, 'r', opener=self._secret_opener) as f:
```

Comment:
This will crash if file doesn't exist. Is that expected?

Reply:
Not a hot code path and we probably want crash here (since will only happen if JWT is missing).